### ISSUE TITLE
Tab completion

### ISSUE THREAD

**Issue by denis-sokolov** [tldr-pages/tldr-node-client#58]
It would be lovely to attach tab completion to this tool, see, for an example, [how npm does it](https://docs.npmjs.com/cli/completion).

**Comment by waldyrious** [comment]
:+1:

**Comment by VarunAgw** [comment]
+1

**Comment by ultrox** [comment]
why not fzf? I made myself super nice fzf-wiget and bound it to `CTRL+O`, which is a key I don't use on terminal, easy to remember `tldr [O]pen`, it automatically puts me into search mode for `linux,common` pages.
I'm on arch, using tldr node client.

```bash
LOC="$HOME/.tldr/cache/pages"

fzf-tldr-widget() {
  local tldrmatch="$(find $LOC/{linux,common} -type f -printf "%f\n" \
  | sed 's/\.md//' \
  | FZF_DEFAULT_OPTS="
     --height ${FZF_TMUX_HEIGHT:-40%} 
     --reverse $FZF_DEFAULT_OPTS" fzf +m)"

  if [[ -z "$tldrmatch" ]]; then
    zle redisplay
    return 0
  fi
  tldr "$tldrmatch"
  local ret=$?
  zle fzf-redraw-prompt
  return $ret
}
zle     -N    fzf-tldr-widget
bindkey '^O'  fzf-tldr-widget
```

**Comment by agnivade** [comment]
There is no harm with fuzzy search. But it is not a replacement for tab completion. Tab completion is a basic feature in many tools; and one which more developers are accustomed to. So we would want that.

**Comment by ultrox** [comment]
@agnivade gotcha, make sense!

**Comment by mearns** [comment]
Is there still interest in this? I see you have zsh-completion, but I've been working with bash-completion lately and would be interested in adding this feature for Hacktoberfest if there's interest from the maintainers.

**Comment by mearns** [comment]
Oh, I just saw there is a rudimentary bash-completion script, but the README hasn't been updated. I'd be interested in updating the bash-completion script to better support the available options (e.g., auto-complete with files after the `--render` option), and update the README to include bash-completion. Please let me know if the maintainers are interested in this.

**Comment by agnivade** [comment]
Nice, thank you @mearns !
